tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/CachedMessageHeadersHelper.cs

```diff
@@ -9,17 +9,25 @@
 using System.IO;
 using System.Reflection;
 using System.Reflection.Emit;
+using System.Text;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Shared;
 using Datadog.Trace.DuckTyping;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNS
 {
-    internal static class CachedMessageHeadersHelper<TMarkerType>
+    internal class CachedMessageHeadersHelper<TMarkerType> : IMessageHeadersHelper
     {
         private const string StringDataType = "Binary";
 
         private static readonly Func<MemoryStream, object> _createMessageAttributeValue;
         private static readonly Func<IDictionary> _createDict;
 
+        public static readonly CachedMessageHeadersHelper<TMarkerType> Instance = new();
```

Comment:
nit: you already have a .cctor() can you create the instance there to be explicit?.